controllers/certificate/certificate.go

```diff
@@ -0,0 +1,185 @@
+package certificate
+
+import (
+	"bytes"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/pem"
+	"fmt"
+	"math"
+	"math/big"
+	"net"
+	"time"
+
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/util/cert"
+	"k8s.io/client-go/util/keyutil"
+	netutils "k8s.io/utils/net"
+)
+
+func generateSelfSignedCertificate(host string, alternateIPs []net.IP, alternateDNS []string, maxAge time.Duration) ([]byte, []byte, error) {
+	validFrom := time.Now().Add(-time.Hour) // valid an hour earlier to avoid flakes due to clock skew
+	caKey, err := rsa.GenerateKey(rand.Reader, 2048)
```

Comment:
There is an [issue in api-gateway](https://github.com/kyma-project/api-gateway/issues/839) that our default kyma gateway cert uses a 2048 bit key. Should we change this key here to use 4096, so there is nothing to complain about this cert?